### PR TITLE
Build new components landing page and simplify component docs templates

### LINKED ISSUE #1088: Intro paragraph for Banner component
New intro paragraph for Banner has been created. See preview [here](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.app.cloud.gov/preview/uswds/uswds-site/jm-banner-intro/components/banner/#site-component-intro) 

- [X] create [draft intro](https://docs.google.com/document/d/1I6bHDbbLz28Hbamr2mL7ZRSSVqe7Wy5vv_iVleKRtSc/edit#heading=h.cwqvrw8xqdb1)
- [x] PR created: https://github.com/uswds/uswds-site/pull/1105
- [x] align w comms plan
- [x] client review 2.17
- [x] update and finalize PR

**DOD**
- [x] all tasked are complete
- [ ] new paragraph is published to banner component page

### LINKED ISSUE #1091: USWDS - Components: Create yml file for component variants.
## Description

Variants don't exist in `_data`, so we can't generate tables for them on component pages. Instead, we have to create them manually.

Settings generate a table based on `yml` files in `_data/settings/components` For example, accordions:
```md
<h5 id="component-settings">Accordion settings</h5>
{% assign settings = site.data.settings.components.accordion %}
{% include settings-table-simple.html settings=settings.contents %}
```

We should do the same for variants.

## Additional information [optional]

Do we create a new `variants` folder in `_data` or rework component settings to take variants as well?

### LINKED ISSUE #1108: USWDS - a11y: Heading order.
## Description

Related to #1104.

**id:** 'heading-order',
**impact:** 'moderate',
**tags:** [ 'cat.semantics', 'best-practice' ],
**description:** 'Ensures the order of headings is semantically correct',
**help:** 'Heading levels should only increase by one',
**helpUrl:** 'https://dequeuniversity.com/rules/axe/3.5/heading-order?application=axeAPI',

### Pages to fix
- [x] Typography
- [x] Tooltip
- [x] Tag
- [x] Table
- [x] Step indicator
- [x] Sidenav
- [x] Search
- [x] Identifier
- [x] Icons
- [x] Header
- [x] Form templates
- [x] Form controls
- [x] Footer
- [x] Data visualizations
- [x] Button groups
- [x] Button
- [x] Breadcrumb
- [x] Banner
- [x] Alert
- [x] Accordion

## Steps to reproduce the issue

1. Open `run-axe.js`
2. Add component URLs from #1104 to `const PAGES` array
4. `npm run axe`

### LINKED ISSUE #1110: USWDS - a11y: Ensure scope attribute is used correctly on tables.
## Description

Related to #1104.

**id:** 'scope-attr-valid',
**impact:** 'moderate',
**tags:** [ 'cat.tables', 'best-practice' ],
**description:** 'Ensures the scope attribute is used correctly on tables',
**help:** 'scope attribute should be used correctly',
**helpUrl:** 'https://dequeuniversity.com/rules/axe/3.5/scope-attr-valid?application=axeAPI',

### Pages to fix
- [ ] accordion
- [ ] alert
- [ ] banner
- [ ] breadcrumb
- [ ] footer
- [ ] form-controls
- [ ] header
- [ ] identifier
- [ ] search
- [ ] sidenav
- [ ] step indicator

## Steps to reproduce the issue

1. Open `run-axe.js`
2. Add component URLs from #1104 to `const PAGES` array
4. `npm run axe`

### PR DESCRIPTION
**Pull request by thisisdano**
~⚠️ HOLD UNTIL AFTER USWDS 2.10.2~
Hold is no longer necessary. Changed base to 2.11.0 release branch.

[Preview](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.app.cloud.gov/preview/uswds/uswds-site/dw-reorganize-components/components/overview/) → 
Paired with https://github.com/uswds/uswds/pull/4056

### Accessibility requirements
- [x] Ensure scope attribute is used correctly on tables. #1110
- [x] Heading order. #1108
- [ ] ~Ensure component page landmarks are unique. #1109~ NOT FIXED HERE

### TODO
- [x] Fix templates pages
- [x] Assure template pages have proper redirect
- [x] Add core compoents notes
- [x] Spacing quirk in date input

Fixes #1091 [Adds variant information to data]
Fixes #1110 [Removes scope from `td` elements]
Fixes #1108 [Heading order]
Fixes #1088 [Banner/identifer core components notes]

- - -

This PR restructures and re-organizes both individual component pages and the components overview page.
- Splits single components overview page into multiple pages: `Components` and `Packages`
- Flattens organization structure, so all components are listed at the top level and there are no categories like `Form controls`
- Adds a filterable list of all components to the overview page
- Adds a `_layouts/component.html` template to power all component pages
- Removes custom markup from individual component pages. Pages are now powered by frontmatter YAML and `[component]/guidance/...` Markdown files

### category
```
category: Components
```

### component
```
component:
  name: accordion
  status: ready
  package: usa-accordion
  dependencies:
``` 

### implementation
```
implementation:
  initProps:
    - property: "`required`"
      element: select
      description: The combo box component will be required in terms of native form validation.
  props:
    - property: "`data-filter`"
      element: .usa-combo-box
      description: The combo box will use this regular expression to filter the combo box options. 
  procedures:
    - procedure: "`disable`"
      parameters: .usa-combo-box
      description: The combo box component will be disabled / read-only.
  variants:
    - variant: "`.usa-accordion--bordered`"
      description: Display a border around accordion content
```

### lead
```
lead: An accordion is a list of headers that hide or reveal additional content when selected.
```

### permalink
```
permalink: /components/accordion/
```

### redirect_from
```
redirect_from:
  - /accordions/
  - /components/accordions/
```

### subnav
```
subnav:
- text: Preview
  href: '#accordion-preview'
- text: Code
  href: '#accordion-code'
- text: Guidance
  href: '#accordion-guidance'
- text: Package
  href: '#accordion-package'
```

### title
```
title: Accordion
```

### tags
```
tags:
  - form
  - forms
  - form controls
  - input
  - dropdown
```

### title
```
title: Combo box
```

### type
```
type: component
```

...

I gave up trying to document it all... :(

### CODE REVIEW

File: _components/accordion/accordion.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+---
+category: Components
+component:
+  name: accordion
+  status: ready
+  package: usa-accordion
+  dependencies:
```
Comment (mejiaj) [inline _components/accordion/accordion.md:7]: What's the intention for `dependencies`? Don't see it used anywhere

File: _components/button-group/guidance/variants/default/accessibility.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+{:.usa-content-list}
```
Comment (mejiaj) [inline _components/button-group/guidance/variants/default/accessibility.md:1]: Is this class required? Not seeing it in previous Accordion or Banner.

File: _components/button-group/button-group.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+layout: styleguide
+lead: A button group collects similar or related actions.
+permalink: /components/button-group/
+redirect_from:
+  - /components/button-groups/
+subnav:
+- text: Default button group
+  href: '#default-button-group'
+- text: Segmented button group
+  href: '#segmented'
```
Comment (mejiaj) [inline _components/button-group/button-group.md:15]: `href: #segmented-button-group`

File: _components/accordion/accordion.md (hunk 1 above)
Comment (thisisdano) [reply _components/accordion/accordion.md:7]: It's in the packages section. Not really used much now, but perhaps more in the future.

File: _components/button-group/guidance/variants/default/accessibility.md (hunk 2 above)
Comment (thisisdano) [reply _components/button-group/guidance/variants/default/accessibility.md:1]: Nope, they're vestiges of some old work. I'll remove.

File: _components/figure/figure.md [hunk 4]
```diff
@@ -1,8 +1,21 @@
 ---
+component:
+  status: alpha
```
Comment (mejiaj) [inline _components/figure/figure.md:3]: Just want to confirm this status prevents it from rendering in the menu, right?

File: _components/button-group/guidance/variants/default/accessibility.md (hunk 2 above)
Comment (thisisdano) [reply _components/button-group/guidance/variants/default/accessibility.md]: Done. Removed others as well with a find/replace

File: _components/button-group/button-group.md (hunk 3 above)
Comment (thisisdano) [reply _components/button-group/button-group.md:15]: FIXED

File: _components/figure/figure.md (hunk 4 above)
Comment (thisisdano) [reply _components/figure/figure.md:3]: Yes, for now they need to have a `status: ready` to appear

File: _components/validation/validation.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    href: '#validation-guidance'
+  - text: Package
+    href: '#validation-package'
+tags:
+  - forms
+  - error
+  - check
+  - checking
+  - checkmark
+---
```
Comment (mejiaj) [inline _components/validation/validation.md:26]: Seeing a visual regression in Validation

## New
image

## Current
image

**Review by mejiaj — COMMENTED** [review]
Code samples seem to be extending beyond bounds of accordion in some components.

You can see this on [Prose](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.app.cloud.gov/preview/uswds/uswds-site/dw-reorganize-components/components/prose/) for example.

image

File: _components/validation/validation.md (hunk 5 above)
Comment (thisisdano) [reply _components/validation/validation.md:26]: FIXED

**Comment by thisisdano** [comment]
OK, I think I've fixed all these issues!

**Review by mejiaj — COMMENTED** [review]
Long settings tables are breaking out of bounds on IE11.

For example Step indicator
image

We could wrap the settings tables with the new `usa-table-container--scrollable`.

Otherwise LGTM

**Pull request merged**